### PR TITLE
Request Response Header Integration

### PR DESCRIPTION
**Pull request by Jha2508**
### Description
- We need to have some specific identifiers for the RPC calls so that we can make sure that the Request made comes from verified source only, hence introducing the concept of `headers`
- Have introduced `RequestHeader` and `ResponseHeader` headers for the RPCs in the `core` package
- These Headers will be taken up by `Interceptors` on `client` and `backend` both to make processing a bit easier for the further layers
- Have introduced a new onboarding RPC `GetCurrentStage` which will give us the current stage of the user in the onboarding journey from `backend`

### CODE REVIEW

File: core/headers.proto [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+package core;
+
+
+option go_package = "Rinnegan/proto-generated/core";
+
+// Request header message
+message RequestHeader {
+    string app_id = 1;       // Unique identifier for your app
+    AppVersion app_version = 2;  // Version of the app
+    string secure_token = 3; // Token to authenticate the request, Planning to take it as JWT Token
```
Comment (Aashirvaad-P) [inline core/headers.proto:14]: Yes, if we want to have a JWT token, we can make a custom Message for that and then accept in that format instead of plain string

**Review by Aashirvaad-P — APPROVED** [review]
Everything looks cool👍🏼

File: core/headers.proto [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Platform platform = 4;     // Platform making the request (e.g., iOS, Android, Web)
+    google.protobuf.Timestamp timestamp = 5;    // Timestamp of the request
+}
+
+// Response header message
+message ResponseHeader {
+    string response_code = 1; // Response code (e.g., 200 for success)
```
Comment (Aashirvaad-P) [inline core/headers.proto]: How is this related to the enum below for ResponseHeader  ?

File: core/headers.proto [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+// AppVersion message with oneof
+message AppVersion {
+    oneof version {
+        int32 version_number = 1;
+        string version_name = 2;
+    }
+}
+
+enum Platform {
```
Comment (Aashirvaad-P) [inline core/headers.proto]: can these enums be put into enums folder if there is no issue with that

File: core/headers.proto (hunk 2 above)
Comment (Jha2508) [reply core/headers.proto]: Sorry missed using the enum here, will change

File: core/headers.proto (hunk 3 above)
Comment (Jha2508) [reply core/headers.proto]: sure , moving to enums file

File: core/headers.proto (hunk 1 above)
Comment (Jha2508) [reply core/headers.proto:14]: A JWT token is a string only according to my understanding, if I am wrong let me know
